blsqpy/mapping.py

```diff
@@ -0,0 +1,67 @@
+import pandas as pd
+import numpy as np
+
+from blsqpy.descriptor import Descriptor
+from collections import OrderedDict
+
+
+def to_expressions(activity, activity_code):
+    expressions = OrderedDict()
+    prefix = ''
+    if activity_code:
+        prefix = activity_code+"_"
+    for state_code, state in Descriptor.as_items(activity.states):
+        for source_code, source in Descriptor.as_items(state.sources):
+            suffix = "_"+source_code
+            if len(source.uids) > 1:
+                state_expressions = []
+                for idx in range(len(source.uids)):
+                    state_expressions.append(
+                        prefix+state_code+"_"+str(idx+1)+suffix)
+                expressions[prefix+state_code+suffix] = state_expressions
+    return expressions
+
+
+def to_mappings(activity, activity_code):
```

Comment:
I don't fully understand at what point we need those, but we should probably refactor into one function ?

Reply:
`to_mappings `
   help translating the previous dataframe format which is the form ou, period, de2.category2, de2.category2,... value
   see the rotated.csv  in the PR
   in more "business" but less brittle format than "just de name"
   ou, period,  activity_state1_source, activity_state2_source, activity_state3_source
   see mapped.csv

`to_expressions`
a last step if you "provided multiple de.category1 in the state.source" is to sum
   eg in the spec the delivered columns so pills_delivered_moh =   ('fSD1ZZo4hTs.oFtySBIkit2', 'pills_delivered_1_moh') +  ('p4GY25ry19H.pQqm0IZ6AJ3', 'pills_delivered_2_moh') +              ('sur5ZhRRQYp.HllvX50cXC0', 'pills_delivered_3_moh'),

   think "dhis2 indicator that just sums multiple category combo"

Reply:
both are used "below" in the map_from_activity(df, activity, activity_code)